pkg/services/correlations/models.go

```diff
@@ -9,7 +9,7 @@ import (
 )
 
 var (
-	ErrSourceDataSourceReadOnly           = errors.New("source data source is read only")
+	ErrCorrelationReadOnly                = errors.New("correlation is read only")
```

Comment:
Nit. Would suggest using the errutil package for declaring errors which enable you to simplify the api.go error handling, see https://github.com/grafana/grafana/blob/main/contribute/backend/errors.md

No blocker for this PR

Reply:
Thanks! I've created a separate task for it to avoid to not add too much to this PR: https://github.com/grafana/grafana/issues/74523